view.go

```diff
@@ -126,29 +131,47 @@ func (v *view) openFragments() error {
 		return errors.Wrap(err, "reading fragments directory")
 	}
 
-	for _, fi := range fis {
-		if fi.IsDir() {
-			continue
-		}
-
-		// Parse filename into integer.
-		shard, err := strconv.ParseUint(filepath.Base(fi.Name()), 10, 64)
-		if err != nil {
-			v.logger.Debugf("WARNING: couldn't use non-integer file as shard in index/field/view %s/%s/%s: %s", v.index, v.field, v.name, fi.Name())
-			continue
-		}
-
-		v.logger.Debugf("open index/field/view/fragment: %s/%s/%s/%d", v.index, v.field, v.name, shard)
-		frag := v.newFragment(v.fragmentPath(shard), shard)
-		if err := frag.Open(); err != nil {
-			return fmt.Errorf("open fragment: shard=%d, err=%s", frag.shard, err)
+	eg, ctx := errgroup.WithContext(context.Background())
+	var mu sync.Mutex
```

Comment:
more descriptive name? `shardMapMu` maybe? (applies to other similar mutexes)

Reply:
i stuck with the local just because it's only used locally on a pretty small scale.